src/plugins/explore/public/application/utils/languages/ppl/get_query_string_with_source/get_query_with_source.ts

```diff
@@ -24,9 +24,9 @@ export const getQueryWithSource = (query: Query): QueryWithQueryAsString => {
 
   let queryStringWithSource: string;
   if (queryString.trim() === '') {
-    queryStringWithSource = `source = ${datasetTitle}`;
+    queryStringWithSource = `source = \`${datasetTitle}\``;
   } else {
-    queryStringWithSource = `source = ${datasetTitle} ${queryString}`;
+    queryStringWithSource = `source = \`${datasetTitle}\` ${queryString}`;
```

Comment:
this might break some data source integrations, can this be added by each data source when constructing the title, rather than always ``?